modules/rosetta/src/test/scala/org/tessellation/rosetta/domain/construction/ConstructionServiceSuite.scala

```diff
@@ -269,33 +278,133 @@ object ConstructionServiceSuite extends MutableIOSuite with Checkers with Transa
+        .map(result => expect.eql(AccountIdentifier(transaction.source, none), result.operations.head.account))
 
     testParseTransactions(isSignedTransaction = false, testCaseCallbackHandler = testCase)
   }
 
   test("parse unsigned transaction returns a negative operation with a negative amount") {
     val testCase = (parseResult: EitherT[F, ConstructionError, ConstructionParse.ParseResult], _: Hashed[Transaction]) =>
       parseResult.rethrowT
-        .map(result => expect.eql(Amount(AmountValue(-1L), currency.DAG).some, result.operations.head.amount))
+        .map(result => expect.eql(Amount(AmountValue(-1L), currency.DAG), result.operations.head.amount))
 
     testParseTransactions(isSignedTransaction = false, testCaseCallbackHandler = testCase)
   }
 
   test("parse unsigned transaction returns a positive operation that gives to the destination address") {
     val testCase = (parseResult: EitherT[F, ConstructionError, ConstructionParse.ParseResult], transaction: Hashed[Transaction]) =>
       parseResult.rethrowT
-        .map(result => expect.eql(AccountIdentifier(transaction.destination, none).some, result.operations.tail.head.account))
+        .map(result => expect.eql(AccountIdentifier(transaction.destination, none), result.operations.tail.head.account))
 
     testParseTransactions(isSignedTransaction = false, testCaseCallbackHandler = testCase)
   }
 
   test("parse unsigned transaction returns a positive operation with a positive amount") {
     val testCase = (parseResult: EitherT[F, ConstructionError, ConstructionParse.ParseResult], _: Hashed[Transaction]) =>
       parseResult.rethrowT
-        .map(result => expect.eql(Amount(AmountValue(1L), currency.DAG).some, result.operations.tail.head.amount))
+        .map(result => expect.eql(Amount(AmountValue(1L), currency.DAG), result.operations.tail.head.amount))
 
     testParseTransactions(isSignedTransaction = false, testCaseCallbackHandler = testCase)
   }
 
+  test("getPayloads returns InvalidPayloadOperations when there's no operation with positive amount") { res =>
+    implicit val (sp, k) = res
+
+    val cs = ConstructionService.make[IO](constSalt)
+
+    val gen: Gen[(Operation, MetadataResult)] =
+      for {
+        (negOp, _) <- payloadOperationsGen
+        metadata <- metadataResultGen
+      } yield (negOp, metadata)
+
+    forall(gen) {
+      case (negOp, metadata) =>
+        cs.getPayloads(NonEmptyList.of(negOp, negOp), metadata)
+          .value
+          .map(expect.eql(Left(InvalidPayloadOperations), _))
+    }
+  }
+
+  test("getPayloads returns InvalidPayloadOperations when one operation amount is not negative of the other") { res =>
+    implicit val (sp, k) = res
+
+    val cs = ConstructionService.make[IO](constSalt)
+
+    def adjustAmountByOne(amt: Amount): Amount = {
+      val newValue = amt.value match {
+        case av if av.isPositive =>
+          if (av.value.value == 1L)
+            av.value.value + 1
```

Comment:
fyi, having imported refined `auto._`, you can compare `av.value === 1L` or probably `av === 1L`. (also, Eq for comparison)

Reply:
Changed

Reply:
@sdavidp are you sure you pushed changes? I see no change 🤔

Reply:
@marcinwadon I think the comments were out of order -- or I replied to the wrong one.. not sure. I had to hit "Comment" button twice earlier to get the comment to go through. Current push has that change, as well as your suggestion about the `InvalidPayloadOperaions`

Reply:
@sdavidp you can comment immediately or start a review (so all your comments are pending) and then you confirm them with another "comment" to publish (as batch). anyway, great. I'll check it out

Reply:
@sdavidp 
> Current push has that change, as well as your suggestion about the `InvalidPayloadOperaions`

there was no push, I still can see no changes 🤔

Reply:
Hmm. This is new; didn't see this, sorry about that.
```
! [remote rejected]   feature/payloads -> feature/payloads (push declined due to email privacy restrictions)
error: failed to push some refs to 'github.com:Constellation-Labs/tessellation.git'
```
I tried to fix notifications. Maybe I broke something. Looking.

Reply:
Succeeded this time. @marcinwadon You should see changes now.